### DISCUSSION TITLE
Sonoff Mini loosing configuration after power off

### DISCUSSION THREAD

**Discussion by epudwjuhc** [arendst/Tasmota#13706] [Q&A] [answered]
Hi

my sonoff mini is loosing the configuration if the power goes down. It comes up back as ESSID tasmota-329222-4642 access point waiting for re-configuration. Then everything works normal again, until power loss then back again to tasmota-329222-4642 AP....

any hints why this would be happening? I assumed the configuration is permanently stored in flash memory?

**Comment by barbudor** [comment, marked as answer]
Settings saving may have been disabled with the 'savedata' command.
What is the current value of 'savedata'?

**Comment by epudwjuhc** [comment]
savedata = ON.

however, the config loss seems to not happen every time the power goes off. Ive tested it right now and config was preserved across power off. yesterday it didn't... weird. Software or hardware bug?

**Reply by barbudor** [reply]
If your power loss is not simple but incluse multiple quick off/on then this is a standard recovery mecanism
Read here : https://tasmota.github.io/docs/Device-Recovery/

Setoption65 1
Can be used to disable this feature

**Reply by epudwjuhc** [reply]
hm, might have been this one. In fact Ive it in a place running with solar power and yesterday the inverter switched on and off several times due to battery undervoltage. 

thanks

**Reply by epudwjuhc** [reply]
in my opinion the interval it uses to detect the recovery procedure is too long! it would be better to have it set to something like max 1 second or so, because as I say solar inverters may restart with a cycle of few seconds (4-5... 10 seconds). Is there a setting (compile time or console) that I can change to reduce this recovery detection interval to something really short like 1 second max?

**Reply by sfromis** [reply]
Due to devices having capacitors holding power for a short time, the recovery interval (how short boots can be till it assumes "ok") is deliberately not brief, and cannot be configured. You can disable it, as already suggested. I'd even say that with an unstable power source, you *should* disable it.

**Reply by epudwjuhc** [reply]
yeah I see what you mean but in my case the interval unfortunately matches the restart interval of solar inverter, so on any fault it will reset tasmota devices that are on that power grid. Disabling should work for me though
